chain/src/chain.rs

```diff
@@ -168,6 +169,41 @@ impl ForkChanges {
             blk.header().number()
         })
     }
+
+    pub fn during_hardfork(&self, hardfork_switch: &HardForkSwitch) -> bool {
+        let hardfork_during_detach =
+            self.check_if_hardfork_during_blocks(hardfork_switch, &self.detached_blocks);
+        let hardfork_during_attach =
+            self.check_if_hardfork_during_blocks(hardfork_switch, &self.attached_blocks);
+
+        hardfork_during_detach || hardfork_during_attach
+    }
+
+    fn check_if_hardfork_during_blocks(
+        &self,
+        hardfork_switch: &HardForkSwitch,
+        blocks: &VecDeque<BlockView>,
+    ) -> bool {
+        if blocks.is_empty() {
+            false
+        } else {
+            // This method assumes that the hardfork epochs are sorted and unique.
```

Comment:
This comment is not right.

It should be:
```suggestion
            // This method assumes that the input blocks are sorted and unique.
```

Focus on the follow code:
```
                let epoch_first = blocks.front().unwrap().epoch().number();
                let epoch_next = blocks
                    .back()
                    .unwrap()
                    .epoch()
                    .minimum_epoch_number_after_n_blocks(1);
                hardfork_epochs.into_iter().any(|hardfork_epoch| {
                    epoch_first < hardfork_epoch && hardfork_epoch <= epoch_next
                })
```

Suppose:
- `hardfork_epochs = [1]`.
- The epoch before `block-15` is `0`; and the epoch of `block-15` is `1`.
- The input blocks are not sorted, they are `[block-12, block-17, block-14]`.

We got:
- The epochs of input blocks are `[epoch-0, epoch-1, epoch-0]`.
- `epoch_first = 0` and `epoch_next = 0`.

Then this method will return `false`.